### ISSUE TITLE
Alacritty 0.13.x immediately closes on macOS

### ISSUE THREAD

**Issue by balsa-sarenac** [alacritty/alacritty#8201]
For bug reports, the following information can help speed up the process. Please
describe the bug that you have found and what you would expect to happen
instead.

When launching alacritty, it briefly appears and immediately closes. This issue appears when I migrate from alacritty 0.12.3 to 0.13.x
Might be helpful: I'm using window manager on macOS. I was working with yabai and later switched to AeroSpace, this issue was present when working with both of them.

### System

OS: macOS 15.0 (had this issue in 14.x as well)
Version: `alacritty 0.13.2 (bb8ea18)` (tested with 0.13.1 and 0.13.0 too)

### Logs

Crashes: STDERR, STDOUT
```
2024-09-26 09:41:26.438 alacritty[39782:508486] +[IMKClient subclass]: chose IMKClient_Legacy
2024-09-26 09:41:26.438 alacritty[39782:508486] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
```
Font/Terminal size: `alacritty -vv`
```
% alacritty -vv                                                                            (trunk)dotfiles
Created log file at "/var/folders/7y/xhcg36l50vj4yfnxvph_8l3w0000gn/T/Alacritty-39811.log"
[0.000002167s] [INFO ] [alacritty] Welcome to Alacritty
[0.000218084s] [INFO ] [alacritty] Version 0.13.2 (bb8ea18)
[0.000554375s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/Users/balsa/.config/alacritty/alacritty.toml"
                                     "/Users/balsa/.config/alacritty/themes/modus_vivendi.toml"
[0.000699584s] [DEBUG] [alacritty] Using environment locale: C/UTF-8/C/C/C/C
[0.018296750s] [INFO ] [alacritty] Using Apple CGL
[0.030707000s] [DEBUG] [alacritty] Picked GL Config:
                                     buffer_type: Some(Rgb { r_size: 8, g_size: 8, b_size: 8 })
                                     alpha_size: 8
                                     num_samples: 0
                                     hardware_accelerated: true
                                     supports_transparency: Some(true)
                                     config_api: Api(OPENGL)
                                     srgb_capable: true
[0.050138334s] [INFO ] [alacritty] Window scale factor: 1
[0.051536292s] [DEBUG] [alacritty] Loading "Iosevka" font
[0.125540500s] [INFO ] [alacritty] Running on Apple M1 Pro
[0.125596042s] [INFO ] [alacritty] OpenGL version 4.1 Metal - 89.3, shader_version 4.10
[0.125606917s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.129700584s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.141812709s] [INFO ] [alacritty] Cell size: 14 x 30
[0.141843209s] [INFO ] [alacritty] Padding: 0 x 0
[0.141855125s] [INFO ] [alacritty] Width: 800, Height: 600
[0.151087167s] [INFO ] [alacritty] PTY dimensions: 20 x 57
[0.153242459s] [INFO ] [alacritty] Initialisation complete
2024-09-26 09:41:47.437 alacritty[39811:509005] +[IMKClient subclass]: chose IMKClient_Legacy
2024-09-26 09:41:47.437 alacritty[39811:509005] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
[0.234949500s] [INFO ] [alacritty] Goodbye
Deleted log file at "/var/folders/7y/xhcg36l50vj4yfnxvph_8l3w0000gn/T/Alacritty-39811.log"
```
Keyboard and bindings: `alacritty --print-events`
```
% alacritty --print-events                                                                 (trunk)dotfiles
Created log file at "/var/folders/7y/xhcg36l50vj4yfnxvph_8l3w0000gn/T/Alacritty-39845.log"
[0.000001917s] [INFO ] [alacritty] Welcome to Alacritty
[0.000273042s] [INFO ] [alacritty] Version 0.13.2 (bb8ea18)
[0.000655500s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/Users/balsa/.config/alacritty/alacritty.toml"
                                     "/Users/balsa/.config/alacritty/themes/modus_vivendi.toml"
[0.019563375s] [INFO ] [alacritty] winit event: NewEvents(Init)
[0.019615208s] [INFO ] [alacritty] winit event: Resumed
[0.019630583s] [INFO ] [alacritty] Using Apple CGL
[0.052133583s] [INFO ] [alacritty] Window scale factor: 1
[0.127362958s] [INFO ] [alacritty] Running on Apple M1 Pro
[0.127413958s] [INFO ] [alacritty] OpenGL version 4.1 Metal - 89.3, shader_version 4.10
[0.127425208s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.143769125s] [INFO ] [alacritty] Cell size: 14 x 30
[0.143826250s] [INFO ] [alacritty] Padding: 0 x 0
[0.143839042s] [INFO ] [alacritty] Width: 800, Height: 600
[0.154028083s] [INFO ] [alacritty] PTY dimensions: 20 x 57
[0.156307250s] [INFO ] [alacritty] Initialisation complete
[0.164424458s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: RedrawRequested }
[0.168121292s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: Resized(PhysicalSize { width: 800, height: 600 }) }
[0.168170917s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: Focused(false) }
[0.168185458s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: RedrawRequested }
[0.169478042s] [INFO ] [alacritty] winit event: AboutToWait
[0.169535000s] [INFO ] [alacritty] winit event: NewEvents(WaitCancelled { start: Instant { tv_sec: 36309, tv_nsec: 522669750 }, requested_resume: Some(Instant { tv_sec: 36309, tv_nsec: 536036166 }) })
2024-09-26 09:42:05.005 alacritty[39845:509422] +[IMKClient subclass]: chose IMKClient_Legacy
2024-09-26 09:42:05.005 alacritty[39845:509422] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
[0.211683000s] [INFO ] [alacritty] winit event: UserEvent(Event { window_id: Some(WindowId(WindowId(5720282544))), payload: Terminal(Wakeup) })
[0.211752125s] [INFO ] [alacritty] winit event: UserEvent(Event { window_id: Some(WindowId(WindowId(5720282544))), payload: Terminal(Wakeup) })
[0.211763583s] [INFO ] [alacritty] winit event: UserEvent(Event { window_id: Some(WindowId(WindowId(5720282544))), payload: Terminal(ChildExit(0)) })
[0.211774667s] [INFO ] [alacritty] winit event: UserEvent(Event { window_id: Some(WindowId(WindowId(5720282544))), payload: Terminal(Exit) })
[0.232157792s] [INFO ] [alacritty] winit event: UserEvent(Event { window_id: Some(WindowId(WindowId(5720282544))), payload: Terminal(Wakeup) })
[0.232215708s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: Focused(true) }
[0.232228458s] [INFO ] [alacritty] winit event: WindowEvent { window_id: WindowId(WindowId(5720282544)), event: Destroyed }
[0.232238375s] [INFO ] [alacritty] winit event: AboutToWait
[0.232736333s] [INFO ] [alacritty] winit event: NewEvents(WaitCancelled { start: Instant { tv_sec: 36309, tv_nsec: 585426208 }, requested_resume: None })
[0.232873708s] [INFO ] [alacritty] winit event: LoopExiting
[0.232951917s] [INFO ] [alacritty] Goodbye
Deleted log file at "/var/folders/7y/xhcg36l50vj4yfnxvph_8l3w0000gn/T/Alacritty-39845.log"
```

**Comment by chrisduerr** [comment]
> I was working with yabai and later switched to AeroSpace, this issue was present when working with both of them.

If you're running a non-standard setup, then figuring this out is up to you.

**Comment by balsa-sarenac** [comment]
Just to clarify then: you're saying that this issue is because window manager resizes/moves alacritty?

**Comment by myrrlyn** [comment]
This occurs to me also on an entirely stock Apple Silicon M2 laptop. I have never installed any other window manager. Running either `alacritty & disown` in a `sh`-compatible shell, or `/usr/bin/open -a /Applications/Alacritty.app`, both cause it to open and persist; it is *only* clicking directly on the app bundle that flashes open and closed. I'm sorry I don't know how to provide any more details than repeating what the original poster provided.

```text
$ alacritty --version
alacritty 0.15.0 (5339553)
$ which alacritty
/opt/homebrew/bin/alacritty
```

macOS version: Sequoia 15.3

<details>
<summary><code>alacritty -vv</code></summary>

```text
Created log file at "/var/folders/9d/5lxt0ntj2vx0w5lkz6lplyvc0000gn/T/Alacritty-64529.log"
[0.000019166s] [INFO ] [alacritty] Welcome to Alacritty
[0.000946791s] [INFO ] [alacritty] Version 0.15.0 (5339553)
[0.004034958s] [INFO ] [alacritty] Configuration files loaded from:
                                     "~/.config/alacritty/alacritty.toml"
                                     "~/.config/alacritty/themes/themes/solarized_osaka.toml"
[0.005265333s] [DEBUG] [alacritty] Using environment locale: en_US.UTF-8
[0.033273541s] [INFO ] [alacritty] Using Apple CGL
[0.051474416s] [DEBUG] [alacritty] Picked GL Config:
                                     buffer_type: Some(Rgb { r_size: 8, g_size: 8, b_size: 8 })
                                     alpha_size: 8
                                     num_samples: 0
                                     hardware_accelerated: true
                                     supports_transparency: Some(true)
                                     config_api: Api(OPENGL)
                                     srgb_capable: true
[0.080092000s] [INFO ] [alacritty] Window scale factor: 2
[0.081330708s] [DEBUG] [alacritty] Loading "Iosevka Nerd Font" font
[0.164365750s] [INFO ] [alacritty] Running on Apple M2
[0.164399583s] [INFO ] [alacritty] OpenGL version 4.1 Metal - 89.3, shader_version 4.10
[0.164433208s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.169405250s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.183666875s] [INFO ] [alacritty] Cell size: 11 x 28
[0.183692083s] [INFO ] [alacritty] Padding: 0 x 0
[0.183703875s] [INFO ] [alacritty] Width: 1600, Height: 1200
[0.222739916s] [INFO ] [alacritty] PTY dimensions: 42 x 145
[0.225300541s] [INFO ] [alacritty] Initialisation complete
[0.242971708s] [INFO ] [alacritty] Font size changed to 30.0 px
[0.243017041s] [INFO ] [alacritty] Cell size: 11 x 28
2025-02-09 09:51:00.328 alacritty[64529:17438859] +[IMKClient subclass]: chose IMKClient_Modern
2025-02-09 09:51:00.328 alacritty[64529:17438859] +[IMKInputSession subclass]: chose IMKInputSession_Modern
[0.290860416s] [INFO ] [alacritty] Padding: 0 x 0
[0.290898916s] [INFO ] [alacritty] Width: 1600, Height: 1200
# as a command-line launch, alacritty stays open until I close it
[4.283260083s] [INFO ] [alacritty] Goodbye
Deleted log file at "/var/folders/9d/5lxt0ntj2vx0w5lkz6lplyvc0000gn/T/Alacritty-64529.log"
```
</details>

<details>
<summary><code>alacritty --print-events</code></summary>

```text
Created log file at "/var/folders/9d/5lxt0ntj2vx0w5lkz6lplyvc0000gn/T/Alacritty-65148.log"
[0.000174375s] [INFO ] [alacritty] Welcome to Alacritty
[0.001115292s] [INFO ] [alacritty] Version 0.15.0 (5339553)
[0.004710417s] [INFO ] [alacritty] Configuration files loaded from:
                                     "~/.config/alacritty/alacritty.toml"
                                     "~/.config/alacritty/themes/themes/solarized_osaka.toml"
[0.040469958s] [INFO ] [alacritty] Using Apple CGL
[0.084241792s] [INFO ] [alacritty] Window scale factor: 2
[0.160247125s] [INFO ] [alacritty] Running on Apple M2
[0.160283458s] [INFO ] [alacritty] OpenGL version 4.1 Metal - 89.3, shader_version 4.10
[0.160313875s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.178953083s] [INFO ] [alacritty] Cell size: 11 x 28
[0.178986417s] [INFO ] [alacritty] Padding: 0 x 0
[0.178999458s] [INFO ] [alacritty] Width: 1600, Height: 1200
[0.203979542s] [INFO ] [alacritty] PTY dimensions: 42 x 145
[0.206686833s] [INFO ] [alacritty] Initialisation complete
[0.213946000s] [INFO ] [alacritty_winit_event] RedrawRequested
[0.220372042s] [INFO ] [alacritty_winit_event] Resized(PhysicalSize { width: 1600, height: 1200 })
[0.220414375s] [INFO ] [alacritty_winit_event] ScaleFactorChanged { scale_factor: 2.0, inner_size_writer: InnerSizeWriter { new_inner_size: (Weak) } }
[0.220433250s] [INFO ] [alacritty_winit_event] Resized(PhysicalSize { width: 1600, height: 1200 })
[0.220446417s] [INFO ] [alacritty_winit_event] Focused(false)
[0.220503333s] [INFO ] [alacritty_winit_event] About to wait
[0.220668833s] [INFO ] [alacritty] Font size changed to 30.0 px
[0.220694542s] [INFO ] [alacritty] Cell size: 11 x 28
[0.227088625s] [INFO ] [alacritty_winit_event] Focused(true)
2025-02-09 09:52:18.714 alacritty[65148:17443099] +[IMKClient subclass]: chose IMKClient_Modern
2025-02-09 09:52:18.714 alacritty[65148:17443099] +[IMKInputSession subclass]: chose IMKInputSession_Modern
[0.297316167s] [INFO ] [alacritty_winit_event] About to wait
[0.298293917s] [INFO ] [alacritty_winit_event] Event { window_id: Some(WindowId(5858482848)), payload: Frame }
[0.298334333s] [INFO ] [alacritty_winit_event] RedrawRequested
[0.304701417s] [INFO ] [alacritty] Padding: 0 x 0
[0.304748458s] [INFO ] [alacritty] Width: 1600, Height: 1200
[0.306286958s] [INFO ] [alacritty_winit_event] About to wait
[0.312649333s] [INFO ] [alacritty_winit_event] Event { window_id: Some(WindowId(5858482848)), payload: Frame }
[0.312696375s] [INFO ] [alacritty_winit_event] About to wait
[0.363901542s] [INFO ] [alacritty_winit_event] Occluded(false)
# stripped the redraws and me typing 'exit\n' ...
[2.933545833s] [INFO ] [alacritty_winit_event] About to wait
[2.933596667s] [INFO ] [alacritty_winit_event] Destroyed
[2.933688708s] [INFO ] [alacritty] Exiting the event loop
[2.933927208s] [INFO ] [alacritty] Goodbye
Deleted log file at "/var/folders/9d/5lxt0ntj2vx0w5lkz6lplyvc0000gn/T/Alacritty-65148.log"
```
</details>

**Comment by archen** [comment]
I stumbled on this thread after upgrading alacritty on macos 15.5. Same behavior/experience as OP with alacritty closing immediately after launch. for me, the most significant setting was

```
[terminal.shell]
program = "/path/to/bin/fish"
```

I started with my full config and pared it down to this point. Without these two lines? no dice. The culprit ended up being an erroneous `SHELL` env var. Worth taking a peek at this for yourselves if you're still experiencing issues.

**Comment by serhiipryusa** [comment]
I have the same issue on MacOS 15.5.
I've just installed Alacritty for the very first time and it just didn't work: closes in ~0.2 seconds after is starts.

@archen you are right. I've added those lines to the config file and now everything works fine.

A bit more details of what I've done just:
1. Created new folder for config: `mkdir -p ~/.config/alacritty`
2. Created new config file there: `touch ~/.config/alacritty/alacritty.toml`
3. Opened this file in Text Edit: `open -a TextEdit ~/.config/alacritty/alacritty.toml`
4. Added the following lines into the config file:
```
[terminal.shell]
program = "/path/to/bin/fish"
```
In my case, the whole path to `fish` is:
```
[terminal.shell]
program = "/opt/homebrew/bin/fish"
```

@archen thank you for your comment!
